pkg/roachpb/span_config.proto

```diff
@@ -151,6 +168,48 @@ message SpanConfigEntry {
   SpanConfig config = 2 [(gogoproto.nullable) = false];
 };
 
+// SystemSpanConfig is a system installed configuration that may apply to
+// multiple spans.
+message SystemSpanConfig {
+  option (gogoproto.equal) = true;
+
+  // ProtectionPolicies is a list of policies which protect data from being
+  // GC-ed.
+  repeated ProtectionPolicy protection_policies = 1 [(gogoproto.nullable) = false];
+}
+
+// SystemSpanConfigTarget is used to specify the target of a SystemSpanConfig.
+message SystemSpanConfigTarget {
```

Comment:
Discussed a bit offline, but could this perhaps be pared down to something like the following?

```proto
message SystemSpanConfigTarget {
   // TenantID indicates the tenant ID of the logical cluster being targeted.
   // For secondary tenants this field is left unset. For the host we can use
   // this field to protect a specific guest tenant. Down in KV we already have
   // access to roachpb.TenantFromContext to disambiguate where the request is
   // originating from -- precisely to avoid needing to add data to our message
   // types to encode as much. We can auth that this field is left empty if
   // originating from a secondary tenant.
   roachpb.TenantID tenant_id = 1 [(gogoproto.customname) = "TenantID", (gogoproto.nullable) = true];
 }
```

A relevant example (not too different from what spanconfig.KVAccessor will need to do):

https://github.com/cockroachdb/cockroach/blob/2569e853358070c83e637de214e4d85a834cacdb/pkg/kv/kvserver/replica_rate_limit.go#L27-L28